src/endpoint.jl

```diff
@@ -21,6 +21,14 @@ RightEndpoint(ep::T, included::Bool) where T = RightEndpoint{T}(ep, included)
 LeftEndpoint(i::AbstractInterval{T}) where T = LeftEndpoint{T}(first(i), first(inclusivity(i)))
 RightEndpoint(i::AbstractInterval{T}) where T = RightEndpoint{T}(last(i), last(inclusivity(i)))
 
+function Base.:(==)(a::LeftEndpoint{T}, b::RightEndpoint{T}) where T
+    a.endpoint == b.endpoint && a.included && b.included
+end
```

Comment:
I think I understand what you're getting at here, but it could definitely use some documentation, because I think most people looking at this code are going to spend some time wondering why this isn't

```julia
a.endpoint == b.endpoint && a.included == b.included
```

Reply:
That minor point aside, this is 👍.